File: app/src/main/java/Database/source/local/VocaDao.java
package Database.source.local;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import Database.Vocabulary;

import static androidx.room.OnConflictStrategy.REPLACE;

@Dao
public interface VocaDao {
    @Insert(onConflict = REPLACE)
    public void insertVocabulary(Vocabulary... vocabularies);

    @Update
    public void updateVocabulary(Vocabulary... vocabularies);

    @Delete
    public void deleteVocabulary(Vocabulary... vocabularies);

    @Query("SELECT * FROM Vocabulary ORDER BY eng ASC")
    public LiveData<List<Vocabulary>> loadAllVocabulary(); // 모든 단어의 영어, 뜻만 load

    @Query("SELECT * from Vocabulary WHERE eng LIKE :eng")
    public LiveData<List<Vocabulary>> loadVocabulary(String eng); // 그 단어의 모든 정보 load
}

File: app/src/main/java/database/source/local/VocaDao.kt
package database.source.local

import androidx.lifecycle.LiveData
import androidx.room.*
import database.Vocabulary

/**
 * DAO class creates actual SQL query.
 */
@Dao
interface VocaDao {
    /**
     * Inserts given vocabularies to the database.
     *
     * @param vocabularies objects to insert to the database
     */
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertVocabulary(vararg vocabularies: Vocabulary?)

    /**
     * Updates given vocabularies.
     *
     * @param vocabularies objects to update from the database
     */
    @Update
    fun updateVocabulary(vararg vocabularies: Vocabulary?)

    /**
     * Deletes given vocabularies from the database.
     *
     * @param vocabularies objects to delete from the database
     */
    @Delete
    fun deleteVocabulary(vararg vocabularies: Vocabulary?)

    /**
     * Loads all vocabulary and sort the result by ascending alphabetic order.
     * @return all vocabulary
     */
    @Query("SELECT * FROM Vocabulary ORDER BY eng ASC")
    fun loadAllVocabulary(): LiveData<MutableList<Vocabulary?>?>?

    /**
     * Loads vocabularies that matches with the english word
     *
     * @param eng english word to search
     * @return vocabularies whose word is similar to eng
     */
    @Query("SELECT * from Vocabulary WHERE eng LIKE :eng")
    fun loadVocabularyByEng(eng: String?): LiveData<MutableList<Vocabulary?>?>?

    /**
     * Loads vocabularies that matches with the korean meaning
     * @param kor korean meaning to search
     * @return vocabularies whose meaning is similar to kor
     */
    @Query("SELECT * from Vocabulary WHERE kor LIKE :kor")
    fun loadVocabularyByKor(kor: String?): LiveData<MutableList<Vocabulary?>?>?
}